modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchInlineResultsPage.kt

```diff
@@ -139,7 +142,21 @@ private fun SearchResultsView(
             }
         }
     }
+    val podcasts by remember { mutableStateOf(state.podcasts) }
+    val podcastsRowState = rememberLazyListState()
+    val episodesRowState = rememberLazyListState()
+    LaunchedEffect(key1 = state.podcasts) {
+        if (state.podcasts.isNotEmpty() && state.podcasts != podcasts) {
```

Comment:
I had to test out removing the mutable state to see what it did. It might be worth adding an explanatory comment making it clear why we need this.

```suggestion
        // Only reset the scroll state if we have podcasts and they are different. This prevents resetting the
        // scroll state when navigating back to the search results after tapping on one of the results.
        if (state.podcasts.isNotEmpty() && state.podcasts != podcasts) {
```